azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,258 @@
+            }
+        }
+
+        if (previousOutput.endpointName !== news.endpointName) {
+            deleteBeforeReplace = true;
+            replaces.push("endpointName");
+        }
+
+        // HTTPS can be enabled/disabled in-place.
+        if (previousOutput.httpsEnabled !== news.httpsEnabled) {
+            changes = true;
+            replaces.push("httpsEnabled");
+        }
+
+        return {
+            deleteBeforeReplace: deleteBeforeReplace,
+            replaces: replaces,
+            changes: changes
+        };
+    }
+
+    async create(inputs: DynamicProviderInputs): Promise<pulumi.dynamic.CreateResult> {
+        const cdnClient = await this.getCDNManagementClient();
+        const result =
+            await cdnClient.customDomains.create(
+                inputs.resourceGroupName,
+                inputs.profileName,
+                inputs.endpointName,
+                this.name,
+                inputs.customDomainHostName);
+
+        if (inputs.httpsEnabled) {
+            console.log("Enabling HTTPS for the custom domain");
+
+            await cdnClient.customDomains.enableCustomHttps(
+                inputs.resourceGroupName,
+                inputs.profileName,
+                inputs.endpointName,
+                this.name);
+        }
+
+        const outs: DynamicProviderOutputs = {
+            ...result,
+            name: this.name,
+            resourceGroupName: inputs.resourceGroupName,
+            profileName: inputs.profileName,
+            endpointName: inputs.endpointName,
+            customDomainHostName: inputs.customDomainHostName,
+            httpsEnabled: inputs.httpsEnabled,
+        };
+
+        return { id: result.id!, outs: outs };
+    }
+
+    async read(id: string, props: DynamicProviderOutputs): Promise<pulumi.dynamic.ReadResult> {
+        const cdnClient = await this.getCDNManagementClient();
+        const result = await cdnClient.customDomains.get(props.resourceGroupName, props.profileName, props.endpointName, this.name);
+        return {
+            id: result.id,
+            props: { ...props, ...result },
+        };
+    }
+
+    async delete(id: string, props: DynamicProviderOutputs): Promise<void> {
+        const cdnClient = await this.getCDNManagementClient();
+        const result = await cdnClient.customDomains.deleteMethod(props.resourceGroupName, props.profileName, props.endpointName, this.name);
+        if (result._response.status >= 400) {
+            throw new Error("Error response received while trying to delete the custom domain.");
+        }
+        if (!result.resourceState) {
+            return;
+        }
+        if (result.resourceState !== 'Deleting') {
+            throw new Error(`Provisioning state of the custom domain was expected to be 'Deleting', but was ${result.resourceState}.`);
+        }
+
+        console.log("The request to delete was successful. However, it can take a minute or two to fully complete deletion.");
```

Comment:
FWIW, the console logging is pretty unusual for a provider. I think it makes more sense for these to be ephemeral log messages (like the k8s status messages), but I can't recall offhand how to do that.

Reply:
`pulumi.log.info()|warn()|...`? I see that those functions accept an ephemeral property. It'll be good to demonstrate that to the reader, that Pulumi has that capability. I'll switch these to that.